src/app/input-bar/input-bar.component.ts

```diff
@@ -18,25 +19,165 @@ export class InputBarComponent implements OnInit {
 
   ngOnInit(): void { }
 
-  pushInput() {
+  pushInput(): void {
     const options = this.parseInput(this.newInput);
-    this.ccdService.updateRegex(options.regex);
-    this.ccdService.updateGroupingKey(options.group);
+    this.ccdService.updateFilters(options.filters);
+    this.ccdService.updateGroupingKey(options.groupBy);
 
-    // we delete any options that are not filter by metadata keys so that
-    // only filter keys are left for the filter pipe to manage.
-    delete options.regex;
-    delete options.group;
-    this.ccdService.updateFilters(options);
+    // We need to scan filters for regex so we can highlight matches
+    options.filters.forEach((filter: FilterOption<any>) => {
+      if (filter.filterKey === FilterKeys.Regex && !filter.negation) {
+        const regex = new RegExp(`(${filter.inputString})`, 'g');
+        this.ccdService.updateRegex(regex);
+      }
+    });
 
   }
 
   parseInput(input: string) {
-    try {
-      return JSON.parse(input);
-    } catch (e) {
-      // By deault we assume user is just inputting a regex
-      return {regex: input};
+    const filters: FilterOption<any>[] = [];
+    let groupBy: GroupByKeys = undefined;
+
+    if (!input) {
+      return {
+        filters,
+        groupBy
+      };
+    }
+    const splitInput = input.trim().split(' ');
+
+    if (splitInput.length === 1) {
+      const newFilter = this.parseArgument(splitInput[0], true);
+      if (newFilter) {
+        filters.push(newFilter);
+      }
+    } else {
+      for (let i = 0; i < splitInput.length; i++) {
+        if (splitInput[i] !== '{') {
+          const newFilter = this.parseArgument(splitInput[i], false);
+          if (newFilter) {
+            filters.push(newFilter);
+          }
+        } else if (i + 2 >= splitInput.length || splitInput[i + 2] !== '}') {
+          // To group we require the syntax '{ grouping_key }'
+          console.error('Invalid input, to group links make sure there are ' +
+              'spaces between the brackets and the grouping key.');
+          return undefined;
```

Comment:
Let's not depend on `undefined` for the default here. 

`undefined` is a sign that something unexpected happen (i.e. we forget to handle thing that is supposed to handle). Since an invalid input is a common thing that we need to handle, how about creating a new type/interface that encapsulate filters, groupBy, and isInputValid ?